venona/pkg/codefresh/codefresh.go

```diff
@@ -92,6 +96,20 @@ func (c cf) Tasks(ctx context.Context) (task.Tasks, error) {
 	return tasks, nil
 }
 
+func (c cf) ReportTaskStatus(ctx context.Context, id task.Id, status task.TaskStatus) error {
+	s, err := status.Marshal()
+	if err != nil {
+		return fmt.Errorf("failed marshalling when reporting task status: %w", err)
+	}
+
+	_, err = c.doRequest(ctx, "POST", bytes.NewBuffer(s), nil, "api", "agent", c.agentID, "tasks", string(id), "statuses")
```

Comment:
is `bytes.NewBuffer(s)` needed, when `s` is already `[]byte`?
maybe it is, i am not sure.

Reply:
It requires `io.Reader` interface, so it seems that it's needed